supplier_product/tests.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from supplier_product.models import SupplierProduct, Product, Supplier
+
+
+class TestSupProduct:
+
+    @pytest.fixture
+    def product(self):
+        prod0 = Product(qr_code="AQMLS5C369VMSL7FOVNX5VLSX0ASSD22", product_name="APPLE", description="sweet!")
+        prod0.save_product()
+        return prod0
+
+    @pytest.fixture
+    def supplier(self):
+        sup0 = Supplier(
+            user_name="ali22",
+            first_name="ali",
+            last_name="ab",
+            password="2345678")
+        sup0.save()
+        return sup0
+
+    @pytest.fixture
+    def supplier_product(self, product, supplier):
+        supprod0 = SupplierProduct(
+            suppler_product_id=123,
+            user_name=supplier,
+            qr_code=product,
+            price=22, quantity=4)
+        supprod0.save()
+        return supprod0
+
+    @pytest.mark.django_db
+    def test_object_saved_in_db(self, supplier_product):
+        assert supplier_product in SupplierProduct.objects.all()
+
+    @pytest.mark.django_db
+    def test_increase_supp_product_quantiy(self, supplier_product):
+        old_quantity = supplier_product.quantity
+        supplier_product.increase_sup_product_quantity()
+        assert supplier_product.quantity == old_quantity + 1
```

Comment:
In here look at meitar logic at increasing quantity and do the same since we have decided to change that and also for the decrease and maybe write a set function for the quantity